extra_data/sourcedata.py

```diff
@@ -102,11 +105,28 @@ def _ipython_key_completions_(self):
         return list(self.keys(inc_timestamps=False))
 
     def _get_first_source_file(self):
-        first_kd = self[self.one_key()]
-
         try:
-            # This property is an empty list if no trains are selected.
+            # May throw TypeError if there are only RUN keys.
+            first_kd = self[self.one_key()]
+
+            # May throw IndexError if no trains are selected.
             sample_path = first_kd.source_file_paths[0]
+
+        except TypeError:
+            for file in self.files:
+                group = file.file['RUN'].get(self.source, None, getlink=True)
+                if group is not None:
+                    break
+            else:
+                # Should this be possible?
+                raise FileStructureError(f'No CONTROL or RUN keys present '
```

Comment:
This error message could be confusing if it occurs on an instrument source. It shouldn't be possible in that case, but maybe we should check `self.is_control` before going into the fallback looking for RUN.